Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs

```diff
@@ -82,12 +96,190 @@ sendPaymentReminderToDriver Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId
+  Payment.OfferListReq
+    { order = offerOrder,
+      customer = Just customerReq,
+      planId = planId.getId,
+      registrationDate = now
+    }
+
+getFinalOrderAmount :: (EncFlow m r, CacheFlow m r, EsqDBFlow m r) => HighPrecMoney -> Id Merchant -> TransporterConfig -> Person -> Id Plan -> HighPrecMoney -> Id DriverFee -> m (HighPrecMoney, Maybe Text)
+getFinalOrderAmount feeWithoutDiscount merchantId transporterConfig driver planId baseAmount driverFeeId = do
+  now <- getCurrentTime
+  let nowLocal = addUTCTime (secondsToNominalDiffTime transporterConfig.timeDiffFromUtc) now
+  if feeWithoutDiscount == 0
+    then do
+      updateStatus CLEARED now driverFeeId
+      return (0, Nothing)
+    else do
+      offers <- Payment.offerList merchantId (makeOfferReq nowLocal feeWithoutDiscount driver planId) -- handle UDFs
+      (finalOrderAmount, offerId) <-
+        if null offers.offerResp
+          then pure (baseAmount, Nothing)
+          else do
+            let bestOffer = minimumBy (comparing (.finalOrderAmount)) offers.offerResp
+            pure (bestOffer.finalOrderAmount, Just bestOffer.offerId)
+      let (platformFee, cgst, sgst) = calculatePlatformFeeAttr finalOrderAmount transporterConfig -- this should be HighPrecMoney
+      updateFee driverFeeId Nothing 0 (round platformFee) cgst sgst now False -- add split logic before update
+      return (finalOrderAmount, offerId)
+
+splitPlatformFee :: HighPrecMoney -> HighPrecMoney -> TransporterConfig -> DriverFee -> [DriverFee]
+splitPlatformFee totalFee maxFeePerEntity transporterConfig DriverFee {..} =
+  let numEntities = totalFee / maxFeePerEntity
+      remainingFee = totalFee `mod'` maxFeePerEntity
+      entityList = replicate (floor numEntities) maxFeePerEntity ++ [remainingFee | remainingFee > 0]
+   in map
+        ( \fee -> do
+            let (platformFee_, cgst, sgst) = calculatePlatformFeeAttr fee transporterConfig
+            DriverFee
+              { platformFee = PlatformFee {fee = round platformFee_, ..},
+                feeType = feeType,
+                ..
+              }
+        )
+        -- govt_charges, num_rides, total_earnings are same for all these
+        entityList
+
+getPlan :: (MonadFlow m, CacheFlow m r) => Maybe DriverPlan -> Id Merchant -> m Plan
+getPlan mbDriverPlan merchantId = do
+  case mbDriverPlan of
+    Just dp -> CQP.findByIdAndPaymentMode dp.planId dp.planType >>= fromMaybeM (PlanNotFound dp.planId.getId)
+    Nothing -> do
+      plans <- CQP.findByMerchantIdAndType merchantId DEFAULT
+      case plans of
+        [] -> throwError $ InternalError "No default plan found"
+        [pl] -> pure pl
+        _ -> throwError $ InternalError "Multiple default plans found"
+
+getFreqAndBaseAmountcase :: PlanBaseAmount -> (Text, HighPrecMoney)
+getFreqAndBaseAmountcase planBaseAmount = case planBaseAmount of
+  PERRIDE_BASE amount -> ("PER_RIDE" :: Text, amount)
```

Comment:
typo `getFreqAndBaseAmountCase`
can we use strict type?
`data PlanBaseFrequency = PER_RIDE | DAILY | WEEKLY | MONTHLY`